src/irmin-pack/IO.ml

```diff
@@ -94,6 +105,10 @@ module Unix : S = struct
   let offset t = t.offset
 
   let force_offset t =
+    (* Raw.Offset.set is called only for {!unsafe_flush}; so this sets [t.offset] to the
+       value of [Raw.Offset.get t.raw], which should be the last offset at which a flush
+       occurred. FIXME it is not clear what this is supposed to do for a read/write
+       instance. Probably this should only be used when [t] is readonly. *) 
```

Comment:
It seems fine to me for `force_offset` to return `t.offset` for read-write instances, in which case the semantics of this function become e.g. "Returns the next free data offset in the underlying file, after synchronising with the file system if necessary". Raising an exception also seems fine.

Reply:
But currently `force_offset` does not return t.offset for read-write instances. Are you suggesting the code is changed?

Reply:
Yes, I think changing the code to do something consistent with the assumptions made elsewhere makes sense.

Reply:
Ioana: Remove this comment, stick to comment in mli